### PR TITLE
Resolve #114

### LINKED ISSUE #114: Flattening build output paths
This comes from #104 - the paths will stay absolute but I will flatten instances of `../` so the paths are lexically equivalent but without all the relative jumps.

### PR DESCRIPTION
**Pull request by Dayvison**

### CODE REVIEW

File: compiler/compiler.go [hunk 1]
```diff
@@ -187,6 +187,11 @@ func CompileWithCommand(cmd *exec.Cmd, workingDir string) (problems types.BuildP
 					problem.File = filepath.Join(workingDir, groups[1])
 				}
 
+				if string(filepath.Separator) != "\\" {
+					problem.File = strings.Replace(problem.File, "\\", "/", -1)
```
Comment (Y-Less) [inline compiler/compiler.go]: Why replace `\` to `/`?  Shouldn't you replace the non-os version to the OS version?  That code is fine on Linux, but most SA:MP code is written on Windows, which should use `\`.

Comment (Dayvison) [reply compiler/compiler.go]: The function [filepath.Clean](https://golang.org/pkg/path/filepath/#Clean) accept both for windows, but only `/` for linux

Comment (Dayvison) [reply compiler/compiler.go]: But thinking now, the condition is no needed, can just replace because windows will accept both

Comment (Southclaws) [reply compiler/compiler.go]: Yeah the replacement is not necessary, Windows apps work fine with `/` and the compiler is no exception. The clean is all that's needed here.

Comment (Dayvison) [reply compiler/compiler.go]: Is necessary on linux

Comment (Southclaws) [reply compiler/compiler.go]: Oh yeah I misread it as the opposite way around!

**Pull request merged**